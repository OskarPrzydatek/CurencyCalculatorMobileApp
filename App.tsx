import React from 'react';
import {Provider as PaperProvider} from 'react-native-paper';
import {NavigationContainer} from '@react-navigation/native';
import { RootNavigator } from './navigation';

const App: React.FC = () => {
  return (
    <NavigationContainer>
      <PaperProvider>
        <RootNavigator />
      </PaperProvider>
    </NavigationContainer>
  );
};

export default App;
