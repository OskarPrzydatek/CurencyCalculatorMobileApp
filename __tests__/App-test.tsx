import {render} from '@testing-library/react-native';
import React from 'react';
import App from '../App';

describe('App', () => {
  test('snapshot', () => {
    const view = render(<App />);
    expect(view).toMatchSnapshot();
  });
});
