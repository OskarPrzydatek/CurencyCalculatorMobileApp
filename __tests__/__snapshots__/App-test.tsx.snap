// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          stackPresentation="push"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="Home"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "gap": 15,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 32,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 40,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      undefined,
                    ]
                  }
                >
                  Currency Calculator
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignSelf": undefined,
                      "backgroundColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 20,
                      "bottom": undefined,
                      "end": undefined,
                      "flex": undefined,
                      "height": undefined,
                      "left": undefined,
                      "opacity": undefined,
                      "position": undefined,
                      "right": undefined,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                      "start": undefined,
                      "top": undefined,
                      "transform": undefined,
                      "width": undefined,
                    }
                  }
                  testID="button-container-outer-layer"
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "rgba(103, 80, 164, 1)",
                        "borderColor": "transparent",
                        "borderRadius": 20,
                        "borderStyle": "solid",
                        "borderWidth": 0,
                        "flex": undefined,
                        "minWidth": 64,
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                      }
                    }
                    testID="button-container"
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          false,
                          {
                            "borderRadius": 20,
                          },
                        ]
                      }
                      testID="button"
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          selectable={false}
                          style={
                            [
                              {
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              {
                                "textAlign": "left",
                              },
                              {
                                "color": "rgba(28, 27, 31, 1)",
                                "writingDirection": "ltr",
                              },
                              [
                                {
                                  "marginHorizontal": 16,
                                  "marginVertical": 9,
                                  "textAlign": "center",
                                },
                                false,
                                {
                                  "marginHorizontal": 24,
                                  "marginVertical": 10,
                                },
                                undefined,
                                false,
                                {
                                  "color": "rgba(255, 255, 255, 1)",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                undefined,
                              ],
                            ]
                          }
                          testID="button-text"
                        >
                          Show Message
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "letterSpacing": 0.15,
                        "lineHeight": 24,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      undefined,
                    ]
                  }
                >
                  version 
                  0.0.1
                </Text>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
